### PR TITLE
telemetry exceptions

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
Updates the telemetry library to handle exceptions as a separate event.
This is exclusive to the core extension, I'll update the rest of the extensions in a separate PR.

### What issues does this PR fix or reference?
@W-6449855@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts [hunk 1]
```diff
@@ -55,13 +55,19 @@ export class SourcePathChecker implements PostconditionChecker<string> {
           return inputs;
         }
       } catch (error) {
-        telemetryService.sendError(error.name);
+        telemetryService.sendException(
+          'force_source_retrieve_with_sourcepath',
+          `Error while parsing oackage directories. ${error.message}`
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts]: *package

File: packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts [hunk 2]
```diff
@@ -26,13 +26,14 @@ export async function getWorkspaceOrgType(
...
-    telemetryService.sendError(
+    telemetryService.sendException(
+      'get_workspace_org_type',
       'workspaceOrgType.getWorkspaceOrgType ran into an undefined username.'
     );
   }
 
   const isScratchOrg = await OrgAuthInfo.isAScratchOrg(username!).catch(err =>
-    telemetryService.sendError(err)
+    telemetryService.sendException('get_workspace_org_type_scratch_org', err)
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts]: Is this sending the error object or the message?

Comment (lcampos) [reply packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts]: Oh right, let me update it to just send a message.

File: packages/salesforcedx-vscode-core/src/telemetry/telemetryReporter.ts [hunk 3]
```diff
@@ -157,6 +159,32 @@ export default class TelemetryReporter extends vscode.Disposable {
     }
   }
 
+  public sendExceptionEvent(
+    exceptionName: string,
+    exceptionMessage: string,
+    measurements?: { [key: string]: number }
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/telemetry/telemetryReporter.ts]: Do we also want the option to send properties in addition to measurements?

Comment (lcampos) [reply packages/salesforcedx-vscode-core/src/telemetry/telemetryReporter.ts]: Eventually yes, that's available in the latest version of application insights but I wanted to keep the changes as small as possible for now. I also didn't see a current need for sending properties so I decided to leave updating the application insights version in a future PR.

**Pull request merged**